cmd/aws-s3-proxy/pkg/serve.go

```diff
@@ -159,11 +162,20 @@ func serve() {
 	}
 
 	// This maps the viper values to the Config object
-	config.Load()
+	config.Load(logger)
 
 	// A gorilla/mux router is used to allow for more control
 	router := mux.NewRouter()
-	router.PathPrefix("/").Handler(common.WrapHandler(controllers.AwsS3)).Methods("GET")
+	router.PathPrefix("/").Handler(common.WrapHandler(controllers.AwsS3Get)).Methods("GET")
+
+	// Require Auth for upload
+	if len(config.Config.BasicAuthPass) > 0 && len(config.Config.BasicAuthUser) > 0 &&
+		config.Config.EnableUpload {
+		router.PathPrefix("/").Handler(common.WrapHandler(controllers.AwsS3Put)).Methods("POST")
+	} else if !(len(config.Config.BasicAuthPass) > 0 && len(config.Config.BasicAuthUser) > 0) &&
+		config.Config.EnableUpload {
+		logger.Fatal("Set up basic auth for upload to work")
+	}
```

Comment:
nit to simplify the log if statements
```suggestion
	// Require Auth for upload
	if (config.Config.EnableUpload) {
	     if len(config.Config.BasicAuthPass) > 0 && len(config.Config.BasicAuthUser) {
	          router.PathPrefix("/").Handler(common.WrapHandler(controllers.AwsS3Put)).Methods("POST")
	     } else {
	          logger.Fatal("Set up basic auth for upload to work")
	    }
	}
```

Reply:
🙌